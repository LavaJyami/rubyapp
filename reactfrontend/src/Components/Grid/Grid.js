import React, { Component } from 'react';
import Square from '../Square/Square';
import '../../App.css';

class Grid extends Component{
  render(){
    const data = this.props.value;
    let values = [];
    for(let i=0; i<data.length;i++){
              let j=i*3;
      values[i] =
         <div className="row" key={i+j}>
            {data[i].map(n => {
              {j++};
              return <Square key={i+j} value={n}/>})}
         </div>
    }
    return(
      <div data-testid = "grid">
        <h1>Word Boggle !!</h1>
        <div className="squarecontainer">
            {values}
        </div>
      </div>
    );
  }
}

export default Grid;
