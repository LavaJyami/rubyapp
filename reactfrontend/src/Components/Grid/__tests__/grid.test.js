import React from 'react';
import ReactDOM from 'react-dom';
import Grid from '../Grid';

import { render, cleanup, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer';


afterEach(cleanup);

it("Mainboard renders without crashing",() => {
  const div = document.createElement("div");
  ReactDOM.render(<Grid value={[["A","B","c","d"],["A","B","c","d"],["A","B","c","d"],["A","B","c","d"]]} />, div);
});

var boardValue = [["A","B","c","d"],["A","B","c","d"],["A","B","c","d"],["A","B","c","d"]];

it("Grid texts and letters are rendered correctly",() => {
  const {getByTestId} = render(<Grid value = {boardValue}/>);
  expect (getByTestId('grid')).toHaveTextContent("Word Boggle !!ABcdABcdABcdABcd");
});

it("matches snapshot",  () => {
  const tree = renderer.create(<Grid value = {boardValue} />).toJSON();
  expect(tree).toMatchSnapshot();
});
