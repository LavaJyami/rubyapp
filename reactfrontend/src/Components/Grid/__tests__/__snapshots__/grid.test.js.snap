// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="grid"
>
  <h1>
    Word Boggle !!
  </h1>
  <div
    className="squarecontainer"
  >
    <div
      className="row"
    >
      <div
        className="square"
        data-testid="square"
      >
        A
      </div>
      <div
        className="square"
        data-testid="square"
      >
        B
      </div>
      <div
        className="square"
        data-testid="square"
      >
        c
      </div>
      <div
        className="square"
        data-testid="square"
      >
        d
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="square"
        data-testid="square"
      >
        A
      </div>
      <div
        className="square"
        data-testid="square"
      >
        B
      </div>
      <div
        className="square"
        data-testid="square"
      >
        c
      </div>
      <div
        className="square"
        data-testid="square"
      >
        d
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="square"
        data-testid="square"
      >
        A
      </div>
      <div
        className="square"
        data-testid="square"
      >
        B
      </div>
      <div
        className="square"
        data-testid="square"
      >
        c
      </div>
      <div
        className="square"
        data-testid="square"
      >
        d
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="square"
        data-testid="square"
      >
        A
      </div>
      <div
        className="square"
        data-testid="square"
      >
        B
      </div>
      <div
        className="square"
        data-testid="square"
      >
        c
      </div>
      <div
        className="square"
        data-testid="square"
      >
        d
      </div>
    </div>
  </div>
</div>
`;
